Sources/Moya/Plugins/NetworkLoggerPlugin.swift

```diff
@@ -62,25 +62,33 @@ private extension NetworkLoggerPlugin {
         return "\(loggerId): [\(date)] \(identifier): \(message)"
     }
 
-    func logNetworkRequest(_ request: URLRequest?) -> [String] {
+    func logNetworkRequest(_ request: RequestType) -> [String] {
 
         var output = [String]()
 
-        output += [format(loggerId, date: date, identifier: "Request", message: request?.description ?? "(invalid request)")]
+        output += [format(loggerId, date: date, identifier: "Request", message: request.request?.description ?? "(invalid request)")]
 
-        if let headers = request?.allHTTPHeaderFields {
+        var headers = request.sessionHeaders
+        if let requestHeaders = request.request?.allHTTPHeaderFields {
+            headers.merge(requestHeaders) { _, requestHeader in
+                return requestHeader
+            }
+        }
+
+        if !headers.isEmpty {
             output += [format(loggerId, date: date, identifier: "Request Headers", message: headers.description)]
         }
 
-        if let bodyStream = request?.httpBodyStream {
+        if let bodyStream = request.request?.httpBodyStream {
```

Comment:
Nit: `request.request` can be a little confuse for people reading this later. What you think?

Reply:
Well yeah, this is from the `RequestType` - we could rename the `request` to `urlRequest` maybe. What do you think?

Reply:
Seems nice 👍